subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultTransformerInvocationFactory.java

```diff
@@ -309,7 +298,8 @@ public TransformationResult call(BuildOperationContext context) {
                     File workspace = executionRequest.getWorkspace();
                     InputChangesInternal inputChanges = executionRequest.getInputChanges().orElse(null);
                     ImmutableList<File> result = transformer.transform(inputArtifactProvider, getOutputDir(workspace), dependencies, inputChanges);
-                    return writeResultsFile(workspace, result);
+                    TransformationResultWriter resultWriter = new TransformationResultWriter(inputArtifact, getOutputDir(workspace));
+                    return resultWriter.writeToFile(getResultsFile(workspace), result);
```

Comment:
Creating an instance of this type just to throw it away immediately seems wrong to me. We could either keep around a long-existing instance, perhaps as a field of the `DefaultTransformerInvocationFactory` itself, and reuse that; or make the whole thing static and pass in all the parameters.

Reply:
I would prefer to leave it as is right now. Let's revisit when we move the building of the transformation result to `DefaultTransformOutputs`.